### PR TITLE
Fix part #180: Hints and solutions Domain layer

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR aims at introducing hints and solutions parts at domain layer. 
For testing purpose I have added dummy hints in TEST_EXPLORATION_ID_5 .

### CODE REVIEW

File: domain/src/main/java/org/oppia/domain/util/StateRetriever.kt [hunk 1]
```diff
@@ -97,6 +103,40 @@ class StateRetriever @Inject constructor(
     return answerGroups
   }
 
+  // Creates the list of hints and solution objects from JSON
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/domain/util/StateRetriever.kt]: How does this deal with Solution? Because in code I am seeing only Hints part.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@veena14cs The test case implementation looks correct to me. But the current json does not deal with solution, so can you add solution to json and updated test case / implementation accordingly.

Also, the final review in this will be from Ben.

File: domain/src/main/java/org/oppia/domain/util/StateRetriever.kt (hunk 1 above)
Comment (veena14cs) [reply domain/src/main/java/org/oppia/domain/util/StateRetriever.kt]: I missed that part. I added it now, please check.

**Review by rt4914 — COMMENTED** [review]
Looks good. Assign this to @BenHenning for final review.

File: model/src/main/proto/exploration.proto [hunk 2]
```diff
@@ -80,7 +80,7 @@ message Interaction {
   repeated AnswerGroup answer_groups = 2;
   Solution solution = 3;
   repeated AnswerGroup confirmed_unclassified_answers = 4;
-  Hint hint = 5;
+  repeated Hint hint = 5;
```
Comment (nikitamarysolomanpvt) [inline model/src/main/proto/exploration.proto:83]: please add comments for variables.

**Review by nikitamarysolomanpvt — APPROVED** [review]
LGTM

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Comment by rt4914** [comment]
@veena14cs Let's not merge this PR for now.
As discussed earlier with @BenHenning its fine we merge the hints and solution implementation on our own. Now for doing that actually it would nice we can introduce entire end-to-end implementation in different PRs so that we can run the app, play with it and that would give a clear idea of how functions should be implemented.

**Comment by veena14cs** [comment]
Duplicate of this PR is created in #866 .

**Pull request closed without merging**